services/src/main/java/org/keycloak/broker/oidc/AbstractOAuth2IdentityProvider.java

```diff
@@ -319,6 +320,16 @@ protected UriBuilder createAuthorizationUrl(AuthenticationRequest request) {
         if (acr != null) {
             uriBuilder.queryParam(OAuth2Constants.ACR_VALUES, acr);
         }
+
+        String nonce = request.getAuthenticationSession().getClientNote(OIDCLoginProtocol.NONCE_PARAM);
+        if (nonce == null || nonce.isEmpty()) {
+            nonce = UUID.randomUUID().toString();
+            request.getAuthenticationSession().setClientNote(OIDCLoginProtocol.NONCE_PARAM, nonce);
+        }
+        if (nonce != null) {
```

Comment:
It couldn't happen that "nonce" will be null if I read correctly? Isn't this "if" check unecessary? Besides that, PR looks good to me.